src/servers/src/http/event.rs

```diff
@@ -250,13 +249,18 @@ pub async fn log_ingester(
 
     let ignore_errors = query_params.ignore_errors.unwrap_or(false);
 
-    let m: mime::Mime = content_type.clone().into();
-    let value = match m.subtype() {
-        mime::JSON => transform_ndjson_array_factory(
+    let value = match content_type {
+        ct if ct == ContentType::json() => transform_ndjson_array_factory(
             Deserializer::from_str(&payload).into_iter(),
             ignore_errors,
         )?,
-        // add more content type support
+        ct if ct == ContentType::text() || ct == ContentType::text_utf8() => Value::Array(
+            payload
+                .lines()
+                .filter(|line| !line.is_empty())
+                .map(|line| json!({ "line": line }))
+                .collect::<Vec<Value>>(),
```

Comment:
Can we modify our pipeline the support line based plain text? Here we transform it to json and later in pipeline we immediately extract this line field, which seems a little waste of compute.

If we want to stick with JSON or JavaScript terms, we can use `this` as field name to reference the whole line.

Reply:
It may take some time to get the pipeline to support plain text. It seems to me that a headerless csv would be better, and then we could use index to refer to field as well.

Reply:
Input of pipeline engine is first converted into [`pipeline value`](https://github.com/GreptimeTeam/greptimedb/blob/849e0b924984727cacfd086a9db5bcb236e17d14/src/pipeline/src/etl/value.rs#L29). Instead of convert plain text into json object, we can directly convert them into `pipeline value`.

@paomian `csv` is yet another format we should support in the future. But plain text is also needed and widely used.

Reply:
@sunng87 @paomian PTAL

Reply:
I'll suggest to add a new variant `TextLines` or just use `String` for pipeline_value. And we can use `this` as field name when referencing this field in pipeline yaml.

Reply:
By allowing pipeline input as `array(string)`, we can eliminate the map here. Please review this commit https://github.com/GreptimeTeam/greptimedb/pull/4300/commits/db0e1ec8e07f5ea211da4de51b33af02d88f45c0

Reply:
@paomian See discussion here. The `Value::String(_)` pattern is used so that `array(string)` can be accepted as input, that we don't need to convert the plain text into a map first.